django/thunderstore/repository/service_account.py

```diff
@@ -0,0 +1,57 @@
+import ulid2
+from django import forms
+from django.contrib.auth.models import User
+from django.core.exceptions import ValidationError
+
+from thunderstore.repository.models import (
+    ServiceAccountMetadata,
+    UploaderIdentity,
+    UploaderIdentityMemberRole,
+)
+
+
+class CreateServiceAccountForm(forms.Form):
+    def __init__(self, user: User, *args, **kwargs) -> None:
+        super().__init__(*args, **kwargs)
+        self.user = user
+        self.fields["identity"] = forms.ModelChoiceField(
+            queryset=UploaderIdentity.objects.filter(members__user=user),
+        )
+
+    def clean_identity(self) -> UploaderIdentity:
+        identity = self.cleaned_data["identity"]
+        member = identity.members.get(user=self.user)
+        if member.role != UploaderIdentityMemberRole.owner:
+            raise ValidationError("Must be identity owner to create a service account")
+        return identity
+
+    def save(self) -> ServiceAccountMetadata:
+        service_account_id = ulid2.generate_ulid_as_uuid()
+        user = User.objects.create_user(service_account_id.hex)
```

Comment:
Not sure about what details to actually give the user. This creates a user with the username of the `ServiceAccountMetadata`'s PK (UUID) and sets an `unusable` password. It might be useful to add a prefix to the username, not sure what kind of filtering would be useful.

Reply:
We do have an unlikely problem with this approach, where we can't guarantee an account with the username doesn't already exist. I would make it possible to name service accounts as well, using the first/last name fields, as these might be nicer to display on the UI than a random UUID.

Is the `unusable` password the same as calling `.set_unusable_password()`?

Reply:
What could be an alternative for a name collision? If we add a prefix to usernames like `sa_` then we could prevent new accounts from registering with this at the start of their name, but with GitHub and Discord integration this might mean having to replace or just remove this prefix when registering them which doesn't seem great.

Renaming service accounts is coming with the management part but I was unsure on what customisations to actually add. A nickname that is unique to that identity's SAs which is just used for the user's first name should be enough but other `User` changes might be needed like `email` (even if this is done in the form and not user changeable), but as far as I'm aware, not setting an email will not break anything.

Yes, when using `create_user`, if you do not pass in a password, it does the same thing as/calls `.set_unusable_password()`

Reply:
I don't think the username collision probability is an issue we have to deal with right now, given that it's astronomically unlikely to happen, and the likeliness of it happening twice is even smaller.

However, usually the convention I've had service accounts follow has been:
1. emails and usernames are the same
2. each service account gets an email generated to them, which we can "hijack" even if an account with that email already exists. This is because we can generate it under a domain we control in a service account namespace. Example: `12839018203.sa@thunderstore.io`. No normal user should have `@thunderstore.io` emails, and if they do, we can force them to rename it to something else.

Ideally we'd restrict users from setting up accounts in the service account naming scheme in the first place, but the oauth integrations make it a little trickier. Also right now we have no guarantees of usernames equating to emails. If you think about it, right now we don't need usernames at all, but we get them anyway thanks to Django.